### ISSUE TITLE
[Feature Request] Shizuku Support as an alternative to root

### ISSUE THREAD

**Issue by realeatham** [NeoApplications/Neo-Backup#832]
**Is your feature request related to a problem? Please describe it.**
This project requires root to access the files to backup, but not everyone has root.

**Describe the primary solution**
Support Shizuku (https://apt.izzysoft.de/fdroid/index/apk/moe.shizuku.privileged.api) to allow the app to use adb instead of root . Shizuku can run on most android devices without root

**Screenshots**
If you're willing to add demonstrative screenshots, you're welcome to do so.

**System Information (please complete the following information):**

- Device: Galaxy S21
- Android Version: Android 14
- ROM: OneUI 6
- App Version: 8.3.6

**Comment by realeatham** [comment]
https://github.com/RikkaApps/Shizuku

**Comment by realeatham** [comment]
GitHub link to shizuku

**Comment by machiav3lli** [comment]
Duplicate #44 and #785

**Comment by extend2to3meters** [comment]
I'm not sure about how others are using Shizuku. But the way I'm using it is to grant temporary root to my device.
I have patched and built a custom rom to forcibly allow `adb root` on a production build (though unmodified userdebug versions would work too). By product, anything that relies on an adb connection can be granted root as well; so I'm able to have Shizuku as my root provider.
Mixplorer has Shizuku support and Termux can use adb so the only missing puzzle piece for me is a nice FOSS backup solution.

Would be nice to have although I am definitely in the fringe minority of root methods.

**Comment by fallsdevil** [comment]
@JustMyGithub I as a user who has no root or intend to have, I use Shizuku for various things, including Macrodroid, Hail, Obtanium, etc. As much as it only performs ADB Shell is still last for these cases even if some apps are not supported. Also, it makes more sense you can do this directly on the device than to have to connect to the PC and run the caming. Anyone I find a viable alternative for me to use apps with features like this, without having to root.

**Comment by JustMyGithub** [comment]
> Macrodroid, Hail, Obtanium

I dont know these, do they have anything to do with Neo Backup or backups in general?

My question is still what a non-root Shizuku can do for Neo Backup, as this is the Neo Backup issue tracker.

**Comment by echeoquehaii** [comment]
Well [SwiftBackup](https://play.google.com/store/apps/details?id=org.swiftapps.swiftbackup&hl=en-US) uses Shizuku to backup external and more data

**Comment by JustMyGithub** [comment]
> Well [SwiftBackup](https://play.google.com/store/apps/details?id=org.swiftapps.swiftbackup&hl=en-US) uses Shizuku to backup external and more data

Sorry, but "external and more" is insanely vague. I guess "external" is the common storage which can be backuped without any tricks at all. "and more" can be everything or nothing. Major issue is that without root you cannot do real app backups. My question 2 years ago was whether Shizuku can work around the root requirement somehow and afaik it can not. It can use root if the device is rooted, but if it is you du not need Shizuku in the first place.
Yes, loopback adb can be useful for some tasks, but not for backups afaik.

**Comment by ifezzz** [comment]
> > Well [SwiftBackup](https://play.google.com/store/apps/details?id=org.swiftapps.swiftbackup&hl=en-US) uses Shizuku to backup external and more data
> 
> Sorry, but "external and more" is insanely vague. I guess "external" is the common storage which can be backuped without any tricks at all. ans "and more" can be evrything or nothing. Major issue is that without root you cannot do real app backups. My question 2 years ago was whether Shizuku can work around the root requirement somehow and afaik it can not. It can use root if the device is rooted, but if it is you du not need Shizuku in teh first place. Yes, loopback adb can be useful for some tasks, but not for backups afaik.

ShizukuPlus: has this feature" Privileged Storage Proxy: Authenticated access to restricted paths like /data/data/ or /data/app/ for backups and file management."

https://github.com/thejaustin/ShizukuPlus/issues/345#issuecomment-5134543225

https://github.com/thejaustin/ShizukuPlus

**Comment by JustMyGithub** [comment]
@ifezzz Tanks for pointing out, I did not know there was a Shizuku+ before. Sadly, as with classic Shizuku, it does not clearly seperate root-,ADB-and new device-owner-capabilities.
As Shizuku claimed beinag able to access /data/data and after hours of reading it turned out that this was limited to root Shizuku, I am sceptic about this one as well. 
In the docs I cannot find anything about root vs. ADB vs. DO (and storage) so I opened an issue with the question over on https://github.com/thejaustin/ShizukuPlus/issues/431
